text/0000-safecoin-alternative-implementation/0000-safecoin-alternative-implementation.md

```diff
@@ -0,0 +1,238 @@
+
+To pay for network resources (e.g. putting data) a `PaymentAccount` must exist where the account's ID is the same as the MAID's public key:
+
+```rust
+struct PaymentAccount {
+    details: CoinAccount,
+    app_rules: BTreeMap<PublicSigningKey, AppRules>,
+}
+
+enum AppRules {
+    Unrestricted,
+    SpendLimit {
+        remaining: Coin,
+    }
+}
+```
+
+The details of the `PaymentAccount` will be explained below at "[Paying for Network Resources](#paying-for-network-resources)".  Until we reach that point, just consider this as a wrapped `CoinAccount` and when `CoinAccount` is mentioned it means either of these types.
+
+
+
+### Farming
+
+Each section of the network will be responsible for a proportion of the total issuable safecoin equal to the proportion of the network address space it manages.  For instance, at the start when the network has a single section covering the full address space, that section will be responsible for all `2^32` safecoin.  When the section splits, each half will be responsible for `2^31` safecoin.
+
+This means that for a section with Prefix length `n`, it will be responsible for `2^(32-n)` safecoin.
+
+Each section's CoinManagers will maintain a record `farmed` (of type `Coin`) of the amount of safecoin farmed at that section.  Any changes to this total will come as a result of an action having passed through Parsec in order to ensure that all managers maintain an eventually consistent record.
+
+The section's `farmed` value will never be allowed to exceed the amount of coins for which that section is responsible.  Ideally no section should ever get close to "running out" of farmable coins; getting the farming rate algorithm correct should ensure that.  However, in the case that a section has farmed all of its coins, it will stop issuing any more until the `farmed` value reduces again.
+
+The actual algorithm defining the farming rate is somewhat orthogonal to this RFC and worthy of its own RFC, so the details will be omitted, but it would seem fairly critical that it ensures sections don't become fully farmed or fully unfarmed.
+
+One benefit of providing coin divisibility is that on a successful farming attempt, it is trivial to pay all farmers in that section rather than the lottery approach taken in the alternative proposal whereby only one farmer is paid at a time, and only if the chosen coin is unfarmed.  Indeed, they could be paid proportional to the value they have provided to the network (this could be measured initially by node age at the moment of payment).  They could also be paid smaller amounts more frequently to encourage new Vault owners to continue to run their Vaults.
+
```

Comment:
Would this then cause issues with nodes that hold more data than others or vice versa? I mean would this code out that possibility?

Reply:
I don't think so.  In fact, because of the coin divisibility and how inexpensive it will be to allocate a farmed amount, I think this gives us more flexibilty to choose a farming rate/farming attempt algorithm which is fairest to all nodes.

As with skipping details of the farming attempt algorithm above, I should probably remove the specific example here mentioning node age, since that's a distraction which doesn't help make the intended point.  I could just as well have mentioned some other metric like data stored/number of successful Get requests/etc.